source/agora/test/ValidatorCleanRestart.d

```diff
@@ -94,13 +94,14 @@ unittest
     // Make all the validators of the set A disable to respond
     set_a.each!(node => node.ctrl.shutdown);
 
-    // give time for the outsiders to be added as validators before sending tx for next block
-    // as catchup for missing txs only occurs after nomination has started
-    Thread.sleep(1.seconds);
-
-    // Block 21 with the new validators in the set B
-    network.generateBlocks(iota(GenesisValidators, allValidators),
-        Height(GenesisValidatorCycle + 1));
+    // Block 21 with the new validators in the set B without relying on tx gossip
+    network.sendTransaction(set_b.front);
+    const enroll_block = set_b.front.getBlock(Height(20));
+    network.expectHeightAndPreImg(iota(GenesisValidators, allValidators), Height(GenesisValidatorCycle + 1),
+        enroll_block.header);
+
+    // assert that all set_b have same block at height 21
+    network.assertSameBlocks(iota(GenesisValidators, allValidators), Height(GenesisValidatorCycle + 1));
```

Comment:
Does the two function calls have different meaning?

Reply:
The first is to generate the block by waiting for the pre-images and then setting the time for the next height.
The second is checking that the block contents are the same.

Reply:
The `expectHeightAndPreImg` already checks the block height for the set B. So I meant the second call looks redundant. But, I will approve it as it's not a big problem soon.